pyscf/nao/gw_iter.py

```diff
@@ -244,11 +244,11 @@ def get_snmw2sf_iter(self, optimize="greedy"):
             for n in range(len(self.nn[s])):    
                 for m in range(self.norbs):
                     # v XVX
-                    a = np.dot(self.kernel_sq, xvx[s][n,m,:])
+                    a = self.kernel_sq.dot(xvx[s][n,m,:])
                     # \chi_{0}v XVX by using matrix vector
                     b = self.gw_chi0_mv(a, self.comega_current)
                     # v\chi_{0}v XVX, this should be equals to bxvx in last approach
-                    a = np.dot(self.kernel_sq, b)
+                    a = self.kernel_sq.dot(b)
```

Comment:
Is it faster?

Reply:
I don't think it is faster, but it is the recommended way, particularly if the matrix is sparse (in this case the dot product use is not from numpy).

Thus it can avoid issues.